### PR TITLE
Use interpolated strings 2 

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Replace `string.Format()` with new `string.Create((IFormatProvider? provider, [InterpolatedStringHandlerArgument(nameof(provider))] ref DefaultInterpolatedStringHandler handler)`.

## PR Context
#18974

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by iSazonov** [comment]
@CarloToso Please resolve merge conflict.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs [hunk 1]
```diff
@@ -248,7 +248,7 @@ private static string GetHalVersion(CimSession session, string systemDirectory)
             try
             {
                 var halPath = CIMHelper.EscapePath(System.IO.Path.Combine(systemDirectory, "hal.dll"));
-                var query = string.Format("SELECT * FROM CIM_DataFile Where Name='{0}'", halPath);
+                var query = string.Format($"SELECT * FROM CIM_DataFile Where Name='{halPath}'");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs:251]:
```suggestion
                var query = string.Create($"SELECT * FROM CIM_DataFile Where Name='{halPath}'");
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs [hunk 2]
```diff
@@ -296,7 +296,7 @@ private void ProcessString(string originalString)
         private static readonly Random _idGenerator = new();
 
         private static string GetGroupLabel(Type inputType)
-            => string.Format("{0} ({1}) <{2:X8}>", inputType.Name, inputType.FullName, _idGenerator.Next());
+            => string.Format($"{inputType.Name} ({inputType.FullName}) <{_idGenerator.Next():X8}>");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs:299]:
```suggestion
            => string.Create($"{inputType.Name} ({inputType.FullName}) <{_idGenerator.Next():X8}>");
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/Group-Object.cs [hunk 3]
```diff
@@ -153,15 +153,15 @@ private static string BuildName(List<ObjectCommandPropertyValue> propValues)
 
                         foreach (object item in propertyValueItems)
                         {
-                            sb.AppendFormat(CultureInfo.InvariantCulture, "{0}, ", item.ToString());
+                            sb.AppendFormat(CultureInfo.InvariantCulture, $"{item}, ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/Group-Object.cs:156]: Please update to Append - ~20 in the PR.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs [hunk 4]
```diff
@@ -2247,13 +2247,13 @@ private string GenerateNewPSSessionOption()
             RunspaceConnectionInfo runspaceConnectionInfo = _remoteRunspaceInfo.Runspace.ConnectionInfo as RunspaceConnectionInfo;
             if (runspaceConnectionInfo != null)
             {
-                result.AppendFormat(null, "-Culture '{0}' ", CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture.ToString()));
-                result.AppendFormat(null, "-UICulture '{0}' ", CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture.ToString()));
+                result.AppendFormat(null, $"-Culture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture.ToString())}' ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:2250]: Please remove ToString()

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs [hunk 5]
```diff
@@ -2247,13 +2247,13 @@ private string GenerateNewPSSessionOption()
...
+                result.AppendFormat(null, $"-UICulture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture.ToString())}' ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:2251]: Please remove ToString()

File: src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfacePrompt.cs [hunk 6]
```diff
@@ -202,7 +202,7 @@ public override
 
                         while (true)
                         {
-                            fieldPromptList.Append(CultureInfo.InvariantCulture, $"{inputList.Count}]: ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostUserInterfacePrompt.cs:205]: Why culture removed?

File: src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs (hunk 1 above)
Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Management/commands/management/GetComputerInfoCommand.cs:251]: We need to add Invariant provider.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs (hunk 4 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs]: Removing ToString()  here breaks stuff

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs (hunk 5 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs]: Removing ToString()  here breaks stuff

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs [hunk 7]
```diff
@@ -2247,13 +2247,13 @@ private string GenerateNewPSSessionOption()
             RunspaceConnectionInfo runspaceConnectionInfo = _remoteRunspaceInfo.Runspace.ConnectionInfo as RunspaceConnectionInfo;
             if (runspaceConnectionInfo != null)
             {
-                result.AppendFormat(null, "-Culture '{0}' ", CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture.ToString()));
-                result.AppendFormat(null, "-UICulture '{0}' ", CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture.ToString()));
+                result.Append(null, $"-Culture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture.ToString())}' ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:2250]:
```suggestion
                result.Append(null, $"-Culture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture)}' ");
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs [hunk 8]
```diff
@@ -2247,13 +2247,13 @@ private string GenerateNewPSSessionOption()
...
+                result.Append(null, $"-UICulture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture.ToString())}' ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:2251]:
```suggestion
                result.Append(null, $"-UICulture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture)}' ");
```

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs [hunk 9]
```diff
@@ -2247,13 +2247,13 @@ private string GenerateNewPSSessionOption()
             RunspaceConnectionInfo runspaceConnectionInfo = _remoteRunspaceInfo.Runspace.ConnectionInfo as RunspaceConnectionInfo;
             if (runspaceConnectionInfo != null)
             {
-                result.AppendFormat(null, "-Culture '{0}' ", CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture.ToString()));
-                result.AppendFormat(null, "-UICulture '{0}' ", CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture.ToString()));
+                result.Append(null, $"-Culture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture)}' ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:2250]:
```suggestion
                result.Append(null, $"-Culture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.Culture.ToString())}' ");
```

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs [hunk 10]
```diff
@@ -2247,13 +2247,13 @@ private string GenerateNewPSSessionOption()
...
+                result.Append(null, $"-UICulture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture)}' ");
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:2251]:
```suggestion
                result.Append(null, $"-UICulture '{CodeGeneration.EscapeSingleQuotedStringContent(runspaceConnectionInfo.UICulture.ToString())}' ");
```

File: src/Microsoft.PowerShell.Commands.Diagnostics/ImportCounterCommand.cs [hunk 11]
```diff
@@ -240,7 +240,7 @@ protected override void EndProcessing()
                     break;
 
                 default:
-                    Debug.Assert(false, string.Format(CultureInfo.InvariantCulture, "Invalid parameter set name: {0}", ParameterSetName));
+                    Debug.Assert(false, $"Invalid parameter set name: {ParameterSetName}");
```
Comment (daxian-dbw) [inline src/Microsoft.PowerShell.Commands.Diagnostics/ImportCounterCommand.cs:243]: Should this be `string.Create(CultureInfo.InvariantCulture, $"...")`?

**Pull request merged**